packages/tiny-graphql-query-compiler/Readme.md

```diff
@@ -0,0 +1,127 @@
+# tiny-graphql-query-compiler
+
+`tiny-graphql-query-compiler` provides a JavaScript-land DSL for constructing GraphQL queries.
+
+Example:
+
+```javascript
+import { compile } from "tiny-graphql-query-compiler";
+
+const query = compile({
+  type: "query",
+  fields: {
+    id: true,
+    name: true,
+    age: true,
+  },
+});
+
+// query =>
+`
+query {
+  id
+  name
+  age
+}
+`;
+```
+
+To pass arguments, you can use the `Call` helper:
+
+```javascript
+import { compile, Call } from "tiny-graphql-query-compiler";
+
+const query = compile({
+  type: "query",
+  fields: {
+    user: Call(
+      { id: 1 },
+      {
+        id: true,
+        name: true,
+        age: true,
+      }
+    ),
+  },
+});
+
+// query =>
+`
+query {
+  user(id: 1) {
+    id
+    name
+    age
+  }
+}
+`;
+```
+
+To pass arguments using GraphQL variables, you can use the `Call` helper with the `Var` helper:
+
+```javascript
+import { compile, Call, Var } from "tiny-graphql-query-compiler";
+
+const query = compile({
+  type: "query",
+  fields: {
+    user: Call(
+      { id: Var({ type: "ID!" }) },
```

Comment:
I'm wondering whether the `required` option should be removed from `Var`. As far as I can tell it only serves to add the `!` onto the type name but as in the example above can just be done in the type name itself. 

Another example is the filter sort options where it is a list of required items, so we are already having to write `!` into the type name already.